hol.mm

```diff
@@ -228,118 +230,150 @@
        the construction of the theorem ` ( A , B ) : bool ` - there is only one
        rule that creates a formula of this type, namely ~ wct , and it requires
        that ` A : bool ` and ` B : bool ` be previously established, so it is
-       safe to reverse the process in ~ wctl and ~ wctr . $)
+       safe to reverse the process in ~ wctl and ~ wctr .  (Contributed by
+       Mario Carneiro, 8-Oct-2014.) $)
     ax-cb1 $a |- R : bool $.
 
-    $( A theorem has type boolean.  (This axiom is unnecessary;
-       see ~ ax-cb1 .) $)
+    $( A theorem has type boolean.  (This axiom is unnecessary; see ~ ax-cb1 .)
+       (Contributed by Mario Carneiro, 8-Oct-2014.) $)
     ax-cb2 $a |- A : bool $.
   $}
 
   ${
     wctl.1 $e |- ( S , T ) : bool $.
     $( Reverse closure for the type of a context.  (This axiom is unnecessary;
-       see ~ ax-cb1 .) $)
-    wctl $a |- S : bool $.
+       see ~ ax-cb1 .)  Prefer ~ wctl .  (New usage is discouraged.)
+       (Contributed by Mario Carneiro, 8-Oct-2014.) $)
+    ax-wctl $a |- S : bool $.
 
     $( Reverse closure for the type of a context.  (This axiom is unnecessary;
-       see ~ ax-cb1 .) $)
-    wctr $a |- T : bool $.
+       see ~ ax-cb1 .)  Prefer ~ wctr .  (New usage is discouraged.)
+       (Contributed by Mario Carneiro, 8-Oct-2014.) $)
+    ax-wctr $a |- T : bool $.
+
+    $( Reverse closure for the type of a context.  (This axiom is unnecessary;
+       see ~ ax-cb1 .)  (Contributed by Mario Carneiro, 8-Oct-2014.) $)
+    wctl $p |- S : bool $=
+      ( ax-wctl ) ABCD $.
```

Comment:
Out of curiosity: why is it necessary to duplicate the $a-statement with an identical $p-statement ?

Reply:
metamath-knife complains that an axiom is not being called `ax-*` or `df-*`. In this database there really is another category, for typing judgments (which are normally excluded from consideration because they are syntax axioms but here have to be provable theorems because they are not context-free). But axiom/definition display is important for the website among other things, so I suppose it's fine to just name them as directed.